src/controllers/followController.js

```diff
@@ -0,0 +1,206 @@
+   * @param {object} response - Response object
+   * @returns {object} An object containing all the data related to the followed user
+   * @memberof FollowController
+   */
+  static async followUser(request, response) {
+    const { username } = request.params;
+    const followingId = request.userData.payload.id;
+    const user = await User.findOne({
+      where: {
+        username
+      }
+    });
+    try {
+      if (user) {
+        if (followingId !== user.id) {
+          Follow.findOrCreate({
+            where: {
+              userId: user.id,
+              followerId: followingId
+            }
+          }).spread((check, created) => {
+            if (!created) {
+              return response.status(409).json(errorResponse([`You are already following ${username}`]));
+            }
+            return response.status(200).json({
+              message: `You are now following ${username}`
+            });
+          });
+        } else {
+          return response.status(409).json(errorResponse(['You cannot follow yourself']));
+        }
+      } else {
+        return response.status(404).json(errorResponse(['User not found']));
+      }
+    } catch (error) {
+      return response.status(500).json(errorResponse([error.message]));
+    }
+  }
+
+  /**
+   * @description Handles the unfollow feature
+   * @param {object} request - Request object
+   * @param {object} response - Response object
+   * @returns {object} An object containing all the data related to the unfollowed user
+   * @memberof FollowController
+   */
+  static async unfollowUser(request, response) {
+    const { username } = request.params;
+    const user = await User.findOne({
+      where: {
+        username
+      }
+    });
+    try {
+      if (user) {
+        const followingId = request.userData.payload.id;
+        if (followingId !== user.id) {
+          const unfollow = await Follow.destroy({
+            where: {
+              userId: user.id,
+              followerId: followingId
+            }
+          });
+          if (unfollow) {
+            return response.status(200).json({
+              message: `You have Unfollowed ${username}`
+            });
+          }
+          return response.status(400).json(errorResponse([`You were not following ${username}`]));
+        }
+        return response.status(409).json(errorResponse(['You cannot unfollow yourself']));
+      }
+      return response.status(404).json(errorResponse(['User not found']));
+    } catch (error) {
+      return response.status(500).json(errorResponse([error.message]));
+    }
+  }
+
+  /**
+     * @description Handles getting all users a user is following
+     * @param {object} request - Request object
+     * @param {object} response - Response object
+     * @returns {Array} An array of object containing required user data
+     * @memberof FollowController
+     */
+  static async getAllFollowing(request, response) {
```

Comment:
@frostyblok I like what you've done with these methods for returning all following and followers of a particular user, KUDOS to you! 👍 . Although, Who is this 'getAllFollowing' and 'getAllFollowers' method for ? Is it to get the following of other users (different from the logged in user), and can a logged in user get all his/her own following and followers? Would he/she also have to search via username to get his/her own following?

Reply:
OK thank you for spotting that! I was following the specification from the user story where a user can get a list of their followings and followers. What do you think?